zerver/lib/message.py

```diff
@@ -164,6 +164,13 @@ def build_message_dict(
                 elif recip['email'] > display_recipient[0]['email']:
                     display_recipient = [display_recipient[0], recip]
 
+        # List all of mentions in the message content
+        ums = UserMessage.objects.filter(message=message_id)
```

Comment:
I feel like we can do this logic more upstream so that we don't have to query UserMessage here.  I haven't looked deeply into this--it's just my hunch.